march_hardware/src/IMotionCube.cpp

```diff
@@ -296,6 +302,11 @@ float IMotionCube::getIMCVoltage()
          static_cast<float>(imc_voltage_iu);  // Conversion to Volt, see Technosoft CoE programming manual
 }
 
+float IMotionCube::getMotorVoltage()
+{
+  return this->read16(this->miso_byte_offsets_.at(IMCObjectName::MotorVoltage)).ui;
```

Comment:
Are you aware that the `.ui` at the end means that the value is read as an unsigned 16 bit integer? I ask this because the return type of the function is `float`. To get the `float` value you would have to do `.f`.

Reply:
This is probably the conversion issue I mention above.

Reply:
You read only 16bit so you cannot make a float out of it. It is a bit strange that they say you have to read 16 bits, but that it is a float. That seems a mistake on their side.